src/components/ContentEditable.tsx

```diff
@@ -45,49 +45,42 @@ export type Props = Omit<
   value?: string;
 };
 
-class OneLineContentEditable extends Component<
-  Props & { forwardedRef: React.ForwardedRef<HTMLElement> }
-> {
-  static defaultProps = {
-    onChange: () => {},
-    onKeyDown: () => {},
-    value: "",
-  };
+function OneLineContentEditable(
+  props: Props & { forwardedRef: React.Ref<HTMLElement> }
+) {
+  const { value, onChange, onKeyDown, forwardedRef, ...rest } = props;
 
-  handleChange = (event: ContentEditableEvent) => {
-    this.props.onChange(event.target.value);
+  const handleChange = (event: ContentEditableEvent) => {
+    onChange && onChange(event.target.value);
   };
 
-  handleKeyDown = (e: React.KeyboardEvent) => {
+  const handleKeyDown = (e: React.KeyboardEvent) => {
     if (e.keyCode === 13 && !e.shiftKey) {
       // ENTER
       e.preventDefault();
     }
     e.stopPropagation();
-    this.props.onKeyDown(e);
+    onKeyDown && onKeyDown(e);
   };
 
-  handlePaste = (ev: React.ClipboardEvent) => {
+  const handlePaste = (ev: React.ClipboardEvent) => {
     ev.preventDefault();
     const text = ev.clipboardData.getData("text");
     document.execCommand("insertText", false, text);
   };
 
-  render() {
-    const { value, onChange, forwardedRef, ...props } = this.props;
-    return (
-      <ContentEditable
-        {...props}
-        html={getInnerHTML(value)}
-        onChange={this.handleChange}
-        tagName="span"
-        spellCheck={false}
-        onKeyDown={this.handleKeyDown}
-        onPaste={this.handlePaste}
-        innerRef={forwardedRef}
-      />
-    );
-  }
+  return (
+    <ContentEditable
+      {...rest}
+      html={getInnerHTML(value ?? "")}
```

Comment:
Whoa. I didn't know the `??` operator existed!

Reply:
[nullish coalescing](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing) is great

Reply:
I'd been using `||` for years, but obviously those semantics are different and I've gotten bitten by that before. `??` is just what I've always wanted!